content/kb/tutorials/databases/aws-s3.md

```diff
@@ -74,17 +76,19 @@ Be sure to replace `xxx` above with the values you noted down earlier, and add t
 
 ## Copy your app secrets to the cloud
 
-As the `secrets.toml` file above is not committed to GitHub, you need to pass its content to your deployed app (on Streamlit Community Cloud) separately. Go to the [app dashboard](https://share.streamlit.io/) and in the app's dropdown menu, click on **Edit Secrets**. Copy the content of `secrets.toml` into the text area. More information is available at [Secrets Management](/streamlit-community-cloud/get-started/deploy-an-app/connect-to-data-sources/secrets-management).
+To host your app on Streamlit Community Cloud, you will need to pass your credentials to your deployed app via secrets. Go to the [app dashboard](https://share.streamlit.io/) and in the app's dropdown menu, click on **Edit Secrets**. Copy the content of `secrets.toml` above into the text area. More information is available at [Secrets Management](/streamlit-community-cloud/get-started/deploy-an-app/connect-to-data-sources/secrets-management).
 
 ![Secrets manager screenshot](/images/databases/edit-secrets.png)
 
-## Add s3fs to your requirements file
+## Add FilesConnection and s3fs to your requirements file
 
-Add the [s3fs](https://github.com/dask/s3fs) package to your `requirements.txt` file, preferably pinning its version (replace `x.x.x` with the version you want installed):
+Add the [FilesConnection](https://github.com/streamlit/files-connection) and [s3fs](https://github.com/dask/s3fs) packages to your `requirements.txt` file, preferably pinning the versions (replace `x.x.x` with the version you want installed):
```

Comment:
Shouldn't the user just have to add files-connection, not both, and assume that files-connection ships with a s3fs in a compatible range?

Reply:
In that case, installing files-connection would also necessarily install the whole boto3 toolchain, (presumably) whole google cloud sdk toolchain, and any others like that we want to support out of the box. I think we should avoid it. WDYT?

I would 1000% support adding some extras-install options for `pip install st-files-connection[s3]`, just haven't had time to add it.

Reply:
Yeah, never mind -- that makes total sense. That would be a great addition once we have time for it, but this is great for now.